test/Azure.Functions.Cli.Tests/E2E/VersionTests.cs

```diff
@@ -24,5 +26,50 @@ await CliTester.Run(new RunConfiguration
                 CommandTimeout = TimeSpan.FromSeconds(30)
             }, _output);
         }
+
+        [Fact]
+        public void CoreToolsAssemblyZipFile_ShouldParseCorrectSegment_WhenValidDownloadLinkIsProvided()
+        {
+            var fakeDownloadLink = "https://example.com/public/coretoolnumber/V4/assemblyfile.zip";
+            var releaseDetail = new CoreToolsRelease { DownloadLink = fakeDownloadLink };
+            var releaseSummary = new ReleaseSummary ("V4",releaseDetail);
+
+            var result = releaseSummary.CoreToolsAssemblyZipFile;
+
+            result.Should().Be("assemblyfile.zip"); // We expect the segment "assemblyfile.zip" based on the provided URL
+        }
+
+        [Fact]
+        public void CoreToolsAssemblyZipFile_ShouldReturnEmpty_WhenDownloadLinkIsNull()
+        {
+            var releaseDetail = new CoreToolsRelease { DownloadLink = null };
+            var releaseSummary = new ReleaseSummary("V4", releaseDetail);
+
+            var result = releaseSummary.CoreToolsAssemblyZipFile; 
+
+            result.Should().Be(string.Empty); // The result should be empty when there is no link
+        }
+
+        [Theory]
+        [InlineData("4.0.6610", "Azure.Functions.Cli.linux-x64.4.0.6610.zip", false)]
+        [InlineData("4.0.1", "Azure.Functions.Cli.linux-x64.4.0.6610.zip", true)]
+
+        public void Test_IsRunningAnOlderVersion(string cliVersion, string latestCoreToolsAssemblyZipFile, bool expected)
+        {
+            bool result = IsRunningAnOlderVersion(cliVersion, latestCoreToolsAssemblyZipFile);
```

Comment:
Let's update the test to validate VersionHelper.IsRunningAnOlderVersion method. We can mock the network call.

Reply:
Please review the latest changes. Added test to validate VersionHelper.IsRunningAnOlderVersion method. Thank you!